src/main/java/org/wso2/carbon/module/swiftiso20022/validation/rules/custom/MT103ArrayFieldValidationRule.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022.validation.rules.custom;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.constants.MT103Constants;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationResult;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationRule;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidatorContext;
+
+import java.util.List;
+
+/**
+ * Validate fields with arrays of strings.
+ * Default case is a repetitive field.
+ * If the field is a multiline field use the constructor with the boolean
+ *
+ */
+public class MT103ArrayFieldValidationRule implements ValidationRule {
+
+    private final List<ValidatorContext> contexts;
+
+    private final boolean isMultilineField;
+
+    // for repetitive fields
```

Comment:
Shall we use a multi-line comment and explain the purpose of the constructor here? Applicable for other places as well. IMO it's better to introduce two classes one for repetitive fields and one for multi-line fields.

Reply:
Add multi-line comments in commit 4e687a247135e6bc9d43274b4f9f97756cf64480. Since both instances contain similar logic to avoid code repetitions both logic is contained in this rule.